src/clr-angular/utils/drag-and-drop/providers/drag-dispatcher.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable, NgZone, Renderer2} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+import {DragPosition} from "../interfaces/client-position";
+import {Draggable} from "../interfaces/draggable";
+
+@Injectable()
+export class DragDispatcher {
+    private _listeners: Function[];
+
+    draggable: Draggable = {self: null};
+
+    private _dragStartPosition: DragPosition;
+    private _dragEndPosition: DragPosition;
+
+    get dragStartPosition() {
+        return this._dragStartPosition;
+    }
+    get dragEndPosition() {
+        return this._dragEndPosition;
+    }
+
+    private _onDragStart: Subject<MouseEvent|TouchEvent> = new Subject<MouseEvent|TouchEvent>();
+    private _onDragMove: Subject<MouseEvent|TouchEvent> = new Subject<MouseEvent|TouchEvent>();
+    private _onDragEnd: Subject<MouseEvent|TouchEvent> = new Subject<MouseEvent|TouchEvent>();
+
+    get onDragStart(): Observable<MouseEvent|TouchEvent> {
+        return this._onDragStart.asObservable();
+    }
+
+    get onDragMove(): Observable<MouseEvent|TouchEvent> {
+        return this._onDragMove.asObservable();
+    }
+
+    get onDragEnd(): Observable<MouseEvent|TouchEvent> {
+        return this._onDragEnd.asObservable();
+    }
+
+    constructor(private _ngZone: NgZone, private _renderer: Renderer2) {}
+
+    initialize() {
```

Comment:
Initialize should receive as arguments everything the class needs to setup. In particular, just give the `Draggable` object here and make the public `draggable` property private.

Reply:
Agree. This would be much better if pass the necessary items as parameters in the `initialize` method. Will address this issue.